diff --git a/sdk/src/levelzero/ze_collector.h b/sdk/src/levelzero/ze_collector.h
index 029f06a..2f8b636 100644
--- a/sdk/src/levelzero/ze_collector.h
+++ b/sdk/src/levelzero/ze_collector.h
@@ -525,7 +525,7 @@ class ZeCollector {
                                     ._queue_handle = queue_handle,
                                     ._device_handle = command.device,
                                     ._phase = phase,
-                                    ._return_code = return_code,
+                                    ._return_code = static_cast<uint32_t>(return_code),
                                     ._correlation_id = command.corr_id_,
                                     ._operation_count = 1,
                                     ._operation_details = op_details};
diff --git a/sdk/src/view_handler.h b/sdk/src/view_handler.h
index ada83be..564fd90 100644
--- a/sdk/src/view_handler.h
+++ b/sdk/src/view_handler.h
@@ -1194,7 +1194,7 @@ inline void CommonSynchEvent(pti_view_record_synchronization& record,
   record._api_id = rec.callback_id_;
   record._event_handle = rec.event_;
   record._number_wait_events = rec.num_wait_events_;
-  record._return_code = rec.result_;
+  record._return_code = static_cast<uint32_t>(rec.result_);
   Instance().InsertRecord(record, record._thread_id);
 }
 
@@ -1328,7 +1328,7 @@ inline void ZeDriverEvent(void* /*data*/, const ZeKernelCommandExecutionRecord&
   record._thread_id = rec.tid_;
   record._process_id = rec.pid_;
   record._api_id = rec.callback_id_;
-  record._return_code = rec.result_;
+  record._return_code = static_cast<uint32_t>(rec.result_);
   record._correlation_id = rec.cid_;
   Instance().InsertRecord(record, record._thread_id);
 }
